src/Adapter/Product/Combination/Repository/CombinationMultiShopRepository.php

```diff
@@ -0,0 +1,669 @@
+     */
+    public function getByShopConstraint(CombinationId $combinationId, ShopConstraint $shopConstraint): Combination
+    {
+        if ($shopConstraint->getShopGroupId()) {
+            throw new InvalidShopConstraintException('Combination has no features related with shop group use single shop and all shops constraints');
+        }
+
+        if ($shopConstraint->forAllShops()) {
+            $shopId = $this->getDefaultShopIdForCombination($combinationId);
+        } else {
+            $shopId = $shopConstraint->getShopId();
+        }
+
+        return $this->get($combinationId, $shopId);
+    }
+
+    /**
+     * @param Combination $combination
+     * @param array $updatableProperties
+     * @param ShopConstraint $shopConstraint
+     * @param int $errorCode
+     */
+    public function partialUpdate(Combination $combination, array $updatableProperties, ShopConstraint $shopConstraint, int $errorCode): void
+    {
+        if ($shopConstraint->getShopGroupId()) {
+            throw new InvalidShopConstraintException('Product combination has no features related with shop group use single shop and all shops constraints');
+        }
+
+        $this->combinationValidator->validate($combination);
+        $combinationId = new CombinationId((int) $combination->id);
+
+        $this->partiallyUpdateObjectModelForShops(
+            $combination,
+            $updatableProperties,
+            $this->getShopIdsByConstraint($combinationId, $shopConstraint),
+            CannotAddCombinationException::class,
+            $errorCode
+        );
+    }
+
+    /**
+     * @param CombinationId $combinationId
+     *
+     * @return ShopId
+     *
+     * @throws ProductNotFoundException
+     */
+    public function getDefaultShopIdForCombination(CombinationId $combinationId): ShopId
+    {
+        $qb = $this->connection->createQueryBuilder();
+
+        $qb
+            ->select('p.id_shop_default')
+            ->from($this->dbPrefix . 'product', 'p')
+            ->leftJoin(
+                'p',
+                $this->dbPrefix . 'product_attribute',
+                'pa',
+                'pa.id_product = p.id_product'
+            )
+            ->where('pa.id_product_attribute = :combinationId')
+            ->setParameter('combinationId', $combinationId->getValue())
+        ;
+
+        $result = $qb->execute()->fetch();
+
+        if (empty($result['id_shop_default'])) {
+            throw new ProductNotFoundException(sprintf(
+                'Could not find Product by combination id %d',
+                $combinationId->getValue()
+            ));
+        }
+
+        return new ShopId((int) $result['id_shop_default']);
+    }
+
+    /**
+     * @param CombinationId $combinationId
+     * @param int $errorCode
+     *
+     * @throws CoreException
+     */
+    public function delete(CombinationId $combinationId, ShopConstraint $shopConstraint, int $errorCode = 0): void
+    {
+        $this->deleteObjectModel(
```

Comment:
Woups, while doing some pair coding with @lartist about images multishop I realized that this delete method is not enough "multishop style" 😅

Although it works fine because you fetch the Combination right on time, so its `id_shop_list` is correctly filled, but that's not how we handle the other methods (add, update) which uses the object model itself as a parameter.

TLDR; I implemented a method in one of my PR that you can copy:
https://github.com/PrestaShop/PrestaShop/blob/051d3ab6da5c24ee1032fe696be8ba228128c922/src/Core/Repository/AbstractMultiShopObjectModelRepository.php#L193

And you should use it here as well I think